src/components/WeatherWidget.vue

```diff
@@ -1,41 +1,93 @@
 <template>
   <div class="weather-widget">
-    <WeatherWidgetPlace
-      v-bind="weatherData.location"
-      :date="selectedDayData?.date ?? 'Today'"
-    />
-    <WeatherWidgetCurrentData v-bind="selectedDayData" />
-    <WeatherWidgetForecast
-      :forecastData="weatherData.forecast"
-      :selectedDay
-      @handleDayChange="handleDayChange"
+    <WeatherWidgetMessage 
+      v-if="messages.length" 
+      :messages 
     />
+    <template v-else-if="selectedDayData && weatherData && locationKeyData">
```

Comment:
I tried to do that with computed property but TypeScript complains about possible undefined/null values for this data when passing to a component so I left it like that. I tried to search for a solution but I didn't find any that fits.

Is there any way of making computed property to satisfy TypeScript after putting it in a condition?

Or should I use optional chaining and handing undefined/null in the components themselves?

Reply:
I believe there is a way to move into the computed property but in this current implementation making Typescript work would be much more verbose. Since this expression is only used once, let's keep this code as is.